### PR TITLE
Custom locator playwright

### PR DESCRIPTION
**Pull request by Georgegriff**
## Motivation/Description of the PR
- Introduce tests for custom locators in webapi across all the helpers
- Work around xpath heuristics in Playwright:  https://github.com/microsoft/playwright/issues/2197
- Resolves issue:  https://github.com/Codeception/CodeceptJS/issues/2314

Applicable helpers:

- [ ] WebDriver
- [ ] Puppeteer
- [ ] Nightmare
- [ ] REST
- [ ] FileHelper
- [ ] Appium
- [ ] Protractor
- [ ] TestCafe
- [x] Playwright

Applicable plugins:

- [ ] allure
- [ ] autoDelay
- [ ] autoLogin
- [x] customLocator
- [ ] pauseOnFail
- [ ] puppeteerCoverage
- [ ] retryFailedStep
- [ ] screenshotOnFail
- [ ] selenoid
- [ ] stepByStepReport
- [ ] wdio

## Type of change

- [ ] :fire: Breaking changes
- [ ] :rocket: New functionality
- [x] :bug: Bug fix
- [ ] :clipboard: Documentation changes/updates
- [ ] :hotsprings: Hot fix
- [ ] :hammer: Markdown files fix - not related to source code
- [ ] :nail_care: Polish code

## Checklist:

- [x] Tests have been added
- [ ] Documentation has been added (Run `npm run docs`)
- [ ] Lint checking (Run `npm run lint`)
- [x] Local tests are passed (Run `npm test`)

### CODE REVIEW

File: lib/helper/Playwright.js [hunk 1]
```diff
@@ -1675,19 +1675,9 @@ class Playwright extends Helper {
...
-      // playwright combined selectors
-      waiter = context.waitForSelector(`${locator.isCustom() ? `${locator.type}=${locator.value}` : locator.simplify()} >> __disabled=false`, { timeout: waitTimeout });
-    } else {
-      const enabledFn = function ([locator, $XPath]) {
-        eval($XPath); // eslint-disable-line no-eval
-        return $XPath(null, locator).filter(el => !el.disabled).length > 0;
-      };
-      waiter = context.waitForFunction(enabledFn, [locator.value, $XPath.toString()], { timeout: waitTimeout });
-    }
+    // playwright combined selectors
```
Comment (Georgegriff) [inline lib/helper/Playwright.js:1679]: Simplified the code path by just using waitForSelector, no need for a special xpath case

File: test/helper/webapi.js [hunk 2]
```diff
@@ -1320,4 +1323,55 @@ module.exports.tests = function () {
       });
     });
   });
+
+  describe('#customLocators', () => {
+    beforeEach(() => {
```
Comment (Georgegriff) [inline test/helper/webapi.js:1328]:
```suggestion
    arfterEach(() => {
```

**Comment by Georgegriff** [comment]
Playwright won't be fixing on their end, but that's fine we can fix here, I need to tidy up linked pr at bit have some failing tests to figure out

**Comment by DavertMik** [comment]
Thank you!

**Pull request merged**